internal/metrics/metrics.go

```diff
@@ -7,65 +7,138 @@ import (
+		},
+	)
+
+	kmmoPreflightResourceNum := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: kmmoPreflightQuery,
+			Help: "Number of existing KMMO Preflights",
+		},
+	)
+
+	kmmoModprobeArgs := prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: kmmoModprobeArgsQuery,
+			Help: "for a given kernel version, describe which modprobe args used (if at all)",
+		},
+		[]string{"name", "namespace", "modprobeArgs"},
+	)
+
+	kmmoModprobeRawArgs := prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Name: completedKMMOStageQuery,
-			Help: "For a given kmmo,namespace, kernel version, stage(device-plugin, module-loader, build), 1 if the stage is completed, 0 if it is not.",
+			Name: kmmoModprobeRawArgsQuery,
+			Help: "for a given kernel version, describe which modprobe raw args used (if at all)",
 		},
-		[]string{"kmmo", "namespace", "kernel", "stage"},
+		[]string{"name", "namespace", "modprobeRawArgs"},
 	)
 
 	return &metrics{
-		kmmoResourcesNum:   kmmoResourcesNum,
-		kmmoCompletedStage: completedStages,
+		kmmoModuleResourcesNum:       kmmoModuleResourcesNum,
+		kmmoInClusterBuildNum:        kmmoInClusterBuildNum,
+		kmmoInClusterSignNum:         kmmoInClusterSignNum,
+		kmmoDevicePluginResourcesNum: kmmoDevicePluginResourcesNum,
+		kmmoPreflightResourceNum:     kmmoPreflightResourceNum,
+		kmmoModprobeArgs:             kmmoModprobeArgs,
+		kmmoModprobeRawArgs:          kmmoModprobeRawArgs,
 	}
 }
 
 func (m *metrics) Register() {
 	runtimemetrics.Registry.MustRegister(
-		m.kmmoResourcesNum,
-		m.kmmoCompletedStage,
+		m.kmmoModuleResourcesNum,
+		m.kmmoInClusterBuildNum,
+		m.kmmoInClusterSignNum,
+		m.kmmoDevicePluginResourcesNum,
+		m.kmmoPreflightResourceNum,
+		m.kmmoModprobeArgs,
 	)
 }
 
-func (m *metrics) SetExistingKMMOModules(value int) {
-	m.kmmoResourcesNum.Set(float64(value))
+func (m *metrics) SetKMMOModulesNum(value int) {
+	m.kmmoModuleResourcesNum.Set(float64(value))
 }
 
-func (m *metrics) SetCompletedStage(kmmoName, kmmoNamespace, kernelVersion, stage string, completed bool) {
-	var value float64
-	if completed {
-		value = 1
-	}
-	m.kmmoCompletedStage.WithLabelValues(kmmoName, kmmoNamespace, kernelVersion, stage).Set(value)
+func (m *metrics) SetKMMOInClusterBuildNum(value int) {
+	m.kmmoInClusterBuildNum.Set(float64(value))
+}
+
+func (m *metrics) SetKMMOInClusterSignNum(value int) {
+	m.kmmoInClusterSignNum.Set(float64(value))
+}
+
+func (m *metrics) SetKMMODevicePluginNum(value int) {
+	m.kmmoDevicePluginResourcesNum.Set(float64(value))
+}
+
+func (m *metrics) SetKMMOPreflightsNum(value int) {
+	m.kmmoPreflightResourceNum.Set(float64(value))
+}
+
+func (m *metrics) SetKMMOModprobeArgs(modName, namespace, modprobeArgs string) {
+	m.kmmoModprobeArgs.WithLabelValues(modName, namespace, modprobeArgs).Set(float64(1))
+}
+
+func (m *metrics) SetKMMOModprobeRawArgs(modName, namespace, modprobeRawArgs string) {
+	m.kmmoModprobeRawArgs.WithLabelValues(modName, namespace, modprobeRawArgs).Set(float64(1))
```

Comment:
Why do we need those functions? What's wrong with using the inner call directly?